src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs

```diff
@@ -23,12 +24,14 @@ internal static class WatsonReporter
         private static ImmutableArray<TelemetrySession> s_telemetrySessions = ImmutableArray<TelemetrySession>.Empty;
         private static ImmutableArray<TraceSource> s_loggers = ImmutableArray<TraceSource>.Empty;
 
+        private static int s_dumpsSubmitted;
+
         public static void InitializeFatalErrorHandlers()
         {
             // Set both handlers to non-fatal Watson. Never fail-fast the ServiceHub process.
             // Any exception that is not recovered from shall be propagated and communicated to the client.
-            var nonFatalHandler = new Action<Exception>(ReportNonFatal);
-            var fatalHandler = nonFatalHandler;
+            var nonFatalHandler = new FatalError.NonFatalHandlerDelegate(ReportNonFatal);
+            var fatalHandler = new Action<Exception>(static (exception) => ReportNonFatal(exception, false));
```

Comment:
```suggestion
            var fatalHandler = new Action<Exception>(static (exception) => ReportNonFatal(exception, forceDump: false));
```

Consider using named argument